### PR TITLE
added UI parameters and osmocom sink

### PR DESCRIPTION
**Pull request by rigid**
To define default values via the UI, I added some <params> to the encoder block.
I also added the osmocom sink and "cleaned" the encoder flowgraph a bit.

### CODE REVIEW

File: lib/encoder_impl.cc [hunk 1]
```diff
@@ -286,6 +287,12 @@ void encoder_impl::set_ps(std::string ps) {
 
 		std::memset(PS, ' ', sizeof(PS));
 		std::memcpy(PS, ps.c_str(), len);
+
+        using std::cout;
+        using std::endl;
+        cout << "PS set to \"" << ps << "\"" << endl;
+        rebuild();
+        rebuild();
```
Comment (bastibl) [inline lib/encoder_impl.cc]: two rebuild()

File: lib/encoder_impl.cc [hunk 2]
```diff
@@ -510,11 +517,10 @@ int encoder_impl::work (int noutput_items,
...
 
-encoder::sptr
-encoder::make (unsigned char pty_locale, int pty, bool ms, std::string ps,
-               double af1, double af2, bool tp, bool ta, int pi_country_code,
-               int pi_coverage_area, int pi_reference_number,
-               std::string radiotext) {
+encoder::sptr encoder::make (unsigned char pty_locale, int pty, bool ms,
+                             std::string ps, double af1, double af2, bool tp,
+                             bool ta, int pi_country_code, int pi_coverage_area,
+                             int pi_reference_number, std::string radiotext) {
```
Comment (bastibl) [inline lib/encoder_impl.cc]: OK,  but generally I would put whitespace stuff in separate commits. AFAIS, the formatting of that function differs from the others either way :-/

File: lib/encoder_impl.cc (hunk 1 above)
Comment (bastibl) [inline lib/encoder_impl.cc]: You indent with spaces, which screws up formatting if one uses a tabwidth != 4

Comment (rigid) [inline lib/encoder_impl.cc]: for a reason. one rebuild() alone doesn't work. (don't ask me why, all this isn't better than a quick hack)

File: apps/rds_tx.grc [hunk 3]
```diff
@@ -1445,6 +1492,10 @@
       <value>14</value>
     </param>
     <param>
+      <key>radiotext</key>
+      <value>GNU Radio &lt;3</value>
+    </param>
+    <param>
       <key>ta</key>
```
Comment (bastibl) [inline apps/rds_tx.grc]: I don't see where this is used. I tried with the loopback flow graph and, at least there, it doesn't seem to work

File: lib/encoder_impl.cc (hunk 1 above)
Comment (rigid) [inline lib/encoder_impl.cc]: I see ndenting with tabs as bad practice since it's not producing same results when working in teams. I didn't find a codingstyle guideline. Feel free to run this through indent and change it to your favourite style.

Comment (bastibl) [inline lib/encoder_impl.cc]: i won't start a discussion about tabs vs. spaces in a comment section on github, but I hope you agree that mixing a lot of very different indentation styles within one file doesn't make much sense.

File: apps/rds_tx.grc (hunk 3 above)
Comment (rigid) [inline apps/rds_tx.grc]: it is used in commit 4a5a22888a ... I admit that the PR is not as clean as it could be, it was a quick hack and I just wanted to share it.

File: lib/encoder_impl.cc [hunk 4]
```diff
@@ -443,7 +464,7 @@ void encoder_impl::prepare_group4a(void) {
 		((toffset>0?0:1)<<5)|((abs(toffset*2))&0x1f);
 }
 
-// for now single-group only
+// TMC Alert-C - http://tinyurl.com/alert-c-coding-handbook
```
Comment (bastibl) [inline lib/encoder_impl.cc]: The full URL might be more stable. (also it looks like this is not pointing where it is supposed to be)

File: grc/rds_encoder.xml [hunk 5]
```diff
@@ -4,7 +4,11 @@
 	<key>gr_rds_encoder</key>
 	<category>RDS</category>
 	<import>import rds</import>
-	<make>rds.encoder($pty_locale)</make>
+	<make>rds.encoder(
+    $pty_locale, $pty, $ms, $ps, $af1, $af2, $tp, $ta,
+    $pi_country_code, $pi_coverage_area, $pi_reference_number, $radiotext
+)</make>
+    <callback>set_ps($ps)</callback>
 	<param>
```
Comment (bastibl) [inline grc/rds_encoder.xml]: Mixed indentation style.

File: lib/encoder_impl.cc (hunk 1 above)
Comment (rigid) [inline lib/encoder_impl.cc]: you brought it up :) but i agree. I didn't even notice it. I should have checked and reconfigured my IDE.

File: lib/encoder_impl.cc (hunk 4 above)
Comment (rigid) [inline lib/encoder_impl.cc]: oh you're right. It should be http://forum.pocketnavigation.de/index.php?page=Attachment&attachmentID=1074401&h=538ea165b7f14bcb4165f989ec406d15eb979118
(although i doubt that that the tinyurl is less stable than the original document. but I get your point of course)

**Comment by bastibl** [comment]
great stuff,  thanks for sharing!
I just tested the changes to the encoder with the loopback flow graph. Works perfectly, except for the alternate frequencies and the radiotext.
Do you have an idea?

**Comment by rigid** [comment]
no, i'm afraid.

**Comment by bastibl** [comment]
OK, I will have a look and merge this PR with some follow-ups soon.

**Comment by rigid** [comment]
cool... glad I could help a bit :)
